src/meta_schedule/utils.h

```diff
@@ -318,14 +318,16 @@ struct ThreadedTraceApply {
 
     for (int i = 0; i < n_; ++i) {
       Item& item = items_[i];
+      bool applied = false;
       try {
-        if (!item.postproc->Apply(sch)) {
-          ++item.fail_counter;
-          return NullOpt;
+        if (item.postproc->Apply(sch)) {
+          applied = true;
         }
       } catch (const std::exception& e) {
-        // Used in multi-thread, only output to screen but failure summary sent to logging
-        LOG(WARNING) << "ThreadedTraceApply::Apply failed with error " << e.what();
+        // left blank intentionally
```

Comment:
One problem here is that we have no visibility into failures.

Would it make sense to catch and log a bounded number before giving up and re-raising?

Reply:
This issue is not caused by a single candidate TIR running multiple times but thousands of them generated by sampler or mutator. They could all be different from each other so only log some number of exceptions is not representative. On the other hand, this is applied in multi-thread, if we keep trace of the logging numbers, it may impact the tuning speed.

Even if we don't log out the exceptions here, we keep track of the failing count, and it's reflected in the log. If the failing count doesn't make sense, it's easy to spot from the log. And we can easily reproduce with running the postprocessor separately once there's any issue. We should be able to see failed tuning of this workload or impacted performance as a result.

One thing we can do to further improve visibility is like I said create a exception counter field in the output, or simply set a check that warns user when there's unexpected postproc failure in the counter. Let me know which one you think is better.

Reply:
"Benign" failures should be caught at the postproc level, e.g. this is how the error is handled for auto tensorization https://github.com/apache/tvm/blob/02c2eae510d6d6c15189427c97819f7ce05f002d/src/meta_schedule/postproc/rewrite_tensorize.cc#L43-L46.

Other errors that propagate to `ThreadedTraceApply` level should be treated as a bug in MS. So I think it is ok to remove this log or this try /catch entirely, and let it crash to reveal the error.